---
title: "miniBulk_EC"
author: "Lin Tian"
date: "5/2/2024"
output: html_document
editor_options: 
  chunk_output_type: console
---

## Load Packages, Data and PCA Plot

```{r}
setwd("~/miniBulk_RNAseq")
library(ggplot2)
library(genefilter)

load("EC.miniBulkRNAseq.RData")

rowVariance_combine <- rowVars(EC.TPM.gene)
names(rowVariance_combine) <- rownames(EC.TPM.gene)
rowVariance_combine <- rowVariance_combine[!grepl("^Rp|^mt-|^Gm", names(rowVariance_combine))]
selected_Rows_combine <- names(sort(rowVariance_combine, decreasing = T)[1:1000])
cx_combine <- sweep(t(EC.TPM.gene[selected_Rows_combine, ]), 2, colMeans(t(EC.TPM.gene[selected_Rows_combine, ])))
sv_combine <- svd(cx_combine)
sv_combine$d^2/sum(sv_combine$d^2) # variance explained
HTseq.EC.ann$PC1_TPM <- sv_combine$u[,1]
HTseq.EC.ann$PC2_TPM <- sv_combine$u[,2]

combine_EC_pca <- ggplot(HTseq.EC.ann) + geom_point(aes(x=PC1_TPM, y=PC2_TPM, fill=Tissue, shape=TumorModel), colour="#000000", size=6) + scale_fill_manual(values=c("#AC0000","#AC0000", "#57A5E5")) + scale_shape_manual(values=c(21,22)) + theme_bw() + theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank(), panel.border=element_blank(), axis.line.x=element_line(colour="black"), axis.line.y=element_line(colour="black")) + coord_fixed(ratio=2.621423/4.82307)
ggsave("~/combine_EC_pca_2.pdf", combine_EC_pca, width=6.28, height=6.28, units="in", useDingbats=FALSE)


```

## Visualize EC signature gene expression

```{r}
library(RColorBrewer)
library(pheatmap)
sinusoid_genes <- c("Lyve1", "Stab1", "F8", "Clec4g", "Stab2")
pheatmap(EC.TPM.gene[sinusoid_genes, c(1,3,5,7,9,11,2,4,10,6,8,12)], scale = "row", clustering_distance_cols="correlation", cluster_cols = FALSE)

capillarized_genes <- c("Esm1", "Sparc", "Col15a1", "Col4a1", "Col4a2", "Itga6", "Lamb1", "Serpinh1", "Nid2")
pheatmap(EC.TPM.gene[capillarized_genes, c(1,3,5,7,9,11,2,4,10,6,8,12)], scale = "row", clustering_distance_cols="correlation", cluster_cols = FALSE)

inflammation_genes <- c("Ly6c1", "Fabp5", "Hmgb2")
pheatmap(EC.TPM.gene[c("Lyve1", "Esm1", "Lamb1", "Lamc1", "Ly6c1"), c(1,3,5,7,9,11,2,4,10,6,8,12)], scale = "row", clustering_distance_cols="correlation", cluster_cols = FALSE)
```

